_posts/2020-08-07-cross-origin-cookies.md

```diff
@@ -0,0 +1,49 @@
+
+In very rare cases, a production DHIS2 application might be running on a different domain than the DHIS2 server. If this issue is affecting a production application in your environment [please let us know as soon as possible by opening a Jira ticket!](https://jira.dhis2.org)
+
+## How can I work around this for application development?
+
+### Using a local DHIS2 instance
+
+By far the most secure way to work around this issue during application development is to run a local instance of DHIS2 against which you can test your application. You can easily spin up one or more DHIS2 instances locally using Docker and the `d2 cluster` command of the [DHIS2 CLI](https://cli.dhis2.nu/#/commands/d2-cluster)
+
+### In Google Chrome or Chromium-based browsers
+
+It is possible to disable the default SameSite=Lax behavior in Chrome and Chromium by setting the ["SameSite by default cookies" flag](chrome://flags/#same-site-by-default-cookies) to **Disabled**. Note that this disables legitimate security behaviors in your browser, so proceed with caution! We recommend that you only disable this flag when actively debugging a DHIS2 application
+
+### In Mozilla Firefox
+
+Firefox stable channel does not yet enforce this cookie policy, so for the moment everything should continue to work. Currently there doesn't appear to be an easy way to disable the policy in Firefox Beta. Stay tuned for updates, and if you find a workaround please share it in the comments below!
+
+## How can I work around this for production applications running on a separate domain?
+
+> **NOTE** Applications installed directly on a DHIS2 application (using the DHIS2 App Management app) will be available on the same domain as the DHIS2 instance, and so **are not affected by this issue**
+
+If a production application is hosted on a separate server and domain than the DHIS2 instance, and it uses cookies for authentication (the most common setup), it **will be affected** by this issue. In order to restore funtionality to external applications it is necessary to set the `SameSite=None-requires-secure` attribute on the DHIS2 authentication cookie and to run the DHIS2 instance with SSL/TLS enabled (so that it is only accessible over HTTPS). DHIS2 does not currently support this behavior by default, so it is necessary to modify the server's cookie responses in a gateway such as NGINX or Apache. If you are hosting a production application outside of your DHIS2 instance and affected by this issue **please reach out to the DHIS2 core team so that we can be made aware of use-cases and can help you work around the issue with a gateway setup**
```

Comment:
Will repeat myself https://github.com/dhis2/dhis2.github.io/pull/37/files#r467694700 just so that we dont miss this one.